### PR TITLE
feat: improve Objc/Swift experience with HiddenFromObjc

### PR DESCRIPTION
**Pull request by buenaflor**
Add `@HiddenFromObjc` annotation to hide `init` with context from Cocoa targets.
This is very useful if we want to have certain things hidden to make the Swift experience better in general

https://kotlinlang.org/docs/native-objc-interop.html

### CODE REVIEW

File: sentry-samples/kmp-app/iosApp/iosApp/iOSApp.swift [hunk 1]
```diff
@@ -3,15 +3,15 @@ import shared
 
 @main
 struct iOSApp: App {
-    let sentry = Sentry()
-
+    let sentry = Sentry.shared
```
Comment (buenaflor) [inline sentry-samples/kmp-app/iosApp/iosApp/iOSApp.swift:6]: Singletons mapped via Kotlin should be accessed via `.shared` or `.companion` or `Companion.shared`: https://kotlinlang.org/docs/native-objc-interop.html#kotlin-singletons

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## 0.0.3
 
+### Features
+
+- Improve Objc/Swift experience with @HiddenFromObjc ([#62](https://github.com/getsentry/sentry-kotlin-multiplatform/pull/62))
```
Comment (marandaneto) [inline CHANGELOG.md:7]: That should be under the ## Unreleased entry

File: sentry-kotlin-multiplatform/src/commonMain/kotlin/io/sentry/kotlin/multiplatform/SentryKMP.kt [hunk 3]
```diff
@@ -19,6 +21,8 @@ object Sentry {
      * @param context: The context (used for retrieving Android Context)
      * @param configuration Options configuration handler.
      */
+    @OptIn(ExperimentalObjCRefinement::class)
+    @HiddenFromObjC
```
Comment (marandaneto) [inline sentry-kotlin-multiplatform/src/commonMain/kotlin/io/sentry/kotlin/multiplatform/SentryKMP.kt:25]: M: Does this require a min. version of KMP that would not work in older versions?
If yes, maybe we can consider just adding a comment that this is not for Apple folks.

**Review by marandaneto — APPROVED** [review]

File: sentry-kotlin-multiplatform/src/commonMain/kotlin/io/sentry/kotlin/multiplatform/SentryKMP.kt (hunk 3 above)
Comment (buenaflor) [reply sentry-kotlin-multiplatform/src/commonMain/kotlin/io/sentry/kotlin/multiplatform/SentryKMP.kt:25]: The min. version for using our KMP sdk is `1.8.0` so nothing would change for current users, I should probably add a badge or something similar to our readme

**Pull request merged**